### ISSUE TITLE
This app not working

### ISSUE THREAD

**Issue by emanabdo7** [aloussase/AlexandriaApp#32]
I can not download any book or convert any book.Please fix this app

**Comment by aloussase** [comment]
Could you tell me a little more about what is not working? What happens when you  try to download a book or convert one?

**Comment by emanabdo7** [comment]
When I try to download a book and click on the download button no thing happen .I clicked many times and nothing downloaded.the same happens when I try to convert a book already exists in my phone and nothing converted.i really don't know why the app not working I hope you can fix this.

**Comment by emanabdo7** [comment]
> Could you tell me a little more about what is not working? What happens when you try to download a book or convert one?

When I try to download a book and click on the download button no thing happen .I clicked many times and nothing downloaded.the same happens when I try to convert a book already exists in my phone and nothing converted.i really don't know why the app not working I hope you can fix this.

**Comment by aloussase** [comment]
I am using the lastest version and can't reproduce the error. It's working fine for me. Are you sure you are using the latest version and have granted all necessary permissions?

**Comment by Phoenix1025** [comment]
Same here! I can find books but cannot download them. It notifies that the app is downloading though. But nothing happens. I checked my permission manager and it says no permission is requested by the app. Fix it bro. This is a helpful app for me because it simplifies my work downloading from library genesis.

**Comment by Phoenix1025** [comment]
> I am using the lastest version and can't reproduce the error. It's working fine for me. Are you sure you are using the latest version and have granted all necessary permissions?

I think thats the problem bro, the app didn't ask for any permission.

**Comment by aloussase** [comment]
Are you checking your Downloads directory? Internet permissions are automatically granted, there is no need to grant them explicitly.

**Comment by emanabdo7** [comment]
> Are you checking your Downloads directory? Internet permissions are automatically granted, there is no need to grant them explicitly.

There are no books downloaded in the download directory. I really don't know why the app is not working. It may need storage permissions, but I checked my permission manager and it says no permission is requested by the app.

**Comment by aloussase** [comment]
I am sorry but I can't reproduce the error on my side. Maybe try uninstalling and installing again.

**Comment by Phoenix1025** [comment]
> > Are you checking your Downloads directory? Internet permissions are automatically granted, there is no need to grant them explicitly.
> 
> There are no books downloaded in the download directory. I really don't know why the app is not working. It may need storage permissions, but I checked my permission manager and it says no permission is requested by the app.

Yes, same here bro. Im on realme c25y android 11 RMX3269_11_A.61. may be its the SAF sh*t of android bro?

**Comment by tedneal** [comment]
I can attest that the app indeed isn't working correctly. Doesn't download anything but search function works fine. Shows file format but file size is grayed out and clicking it does nothing.

On a rooted Unihertz Titan running Android 10.

**Comment by zetaomegagon** [comment]
1. @aloussase I think this issue should focus on the downloads problem. If you are okay with that then @emanabdo7 should retitle the issue to "Downloads not working" or similar

2. How are people testing downloads?

@tedneal @Phoenix1025 @Delorean12DMC 

Here's how I tested:

A) open the app
B) open Rethink DNS app log
C) do a search
D) check DNS logs and see contact to `library.lol`
E) try to download a book, and see that the Download Manager is connecting to `download.library.lol`
F) check for a download notification...nothing
G) check downloads...nothing
H) search for Common Lisp
I) hit the download button on 10 books
J) repeat step F...after a bit I see lots of "Download Failed messages"; one or two downloads in progress
K) didn't see or hear the notification. Why? Check Download Manager notification settings. I can't change them (OnePlus 9 Pro, DivestOS 20), so download notifications are silent

3. I suggest people test like I did since downloads from LibGen, in my experience, can be pretty flakey

**Comment by tedneal** [comment]
It doesn't request storage access at least in my case. This might be the reason downloading isn't an option. Also can't upload anything to convert to another format from internal storage.

**Comment by aloussase** [comment]
I just noticed the conversion service was down. That would be causing problems with conversions, but it is already fixed.

**Comment by aloussase** [comment]
For everyone having problems I suggest deleting the app and installing again, you should be asked notification permissions for example. I've been using the latest version to download books successfully so I don't really know what may be causing you guys issues.

**Comment by tedneal** [comment]
![Screenshot_20240608-143937](https://github.com/aloussase/AlexandriaApp/assets/88007821/49f73b65-8901-469e-abf0-93e51c3d09c0)
![Screenshot_20240608-143908](https://github.com/aloussase/AlexandriaApp/assets/88007821/1488165f-f269-4954-a132-6a20ccc840d2)

This is what I see on the app. There's not anywhere to access internal storage to convert the file. Also downloadable files are grayed out as shown.

**Comment by aloussase** [comment]
What do you mean downloadable files are greyed out? I can see the colored buttons, you need to click on those to download the book in the specified file format. The convert books screen does look weird on your device. You should be able to scroll down to see the file choose button below the conversion format selectors. What device are you using?

**Comment by tedneal** [comment]
It is a Unihertz Titan. It has a 1:1 aspect ratio. I cannot find a way to scroll down. The file size is grayed out. Clicking the colored buttons seems to do nothing. Nothing has appeared in downloads.

**Comment by aloussase** [comment]
Does that device have a touch screen? Or is it just handled through the keyboard?

**Comment by aloussase** [comment]
Im gonna try that device in the emulator, though I doubt it's available lol

**Comment by tedneal** [comment]
> Does that device have a touch screen? Or is it just handled through the keyboard?

Yes, it has a touch screen and it also has a touch sensitive keyboard for scrolling.

**Comment by tedneal** [comment]
> Im gonna try that device in the emulator, though I doubt it's available lol

It might not be available. It's a very niche device.

**Comment by zetaomegagon** [comment]
@tedneal it should only need to request storage access if the app needs general access to storage. Did you try the method I outlined? Sometimes books just don't download from LibGen *outside* the application

**Comment by tedneal** [comment]
> @tedneal it should only need to request storage access if the app needs general access to storage. Did you try the method I outlined? Sometimes books just don't download from LibGen _outside_ the application

I have followed your method. It connects to library.lol but I don't see any connections to download.library.lol

Nothing appears.

**Comment by zetaomegagon** [comment]
@tedneal well Download Manager connects to download.library.lol

Anyway, here's a screen recording of what I see

https://streamable.com/n9c6sm

**Comment by aloussase** [comment]
Mmmm that makes me think the problem might be with the library genesis mirror being used. Can you verify that from your location you can download books from libgen.is?

**Comment by tedneal** [comment]
> Mmmm that makes me think the problem might be with the library genesis mirror being used. Can you verify that from your location you can download books from libgen.is?

I don't seem to have issues accessing libgen.is but the downloads are accessing one of two mirrors. The attached screenshot shows one failing. The other url of library.lol seems to connect fine.
![Screenshot_20240608-174529](https://github.com/aloussase/AlexandriaApp/assets/88007821/abfa8447-f7fd-4db5-a116-11732180b0ca)

**Comment by aloussase** [comment]
Maybe I can make this a setting, choosing which libgen mirror to use.

**Comment by emanabdo7** [comment]
I think that this app is not working on some mobile brands like Huawei. My phone is Huawei, and the app is not working. However, when I tested it on Oppo and Samsung, it's okay.
